### DISCUSSION TITLE
Integration test mediatr handlers BUT populate HttpContext

### DISCUSSION THREAD

**Discussion by thatdotnetguy** [LuckyPennySoftware/MediatR#836] [Q&A] [answered]
I'm setting up some integration tests for mediatr handlers using xunit, respawn, and webapplicationfactory....

One of classes ultimately called by the mediatr handler that is being tested.. has a dependency on IHttpContextAccessor as you can see below

![image](https://user-images.githubusercontent.com/63023163/218364794-aeabe885-5c96-4ca8-8f77-a4692e3fdc00.png)

thought I could do AddSingleton for the webapplicationfactory....

![image](https://user-images.githubusercontent.com/63023163/218364853-33779647-a9c9-4546-b892-77804bfe0371.png)

But when the integration test runs.. the httpContextAccessor is not null but the **HttpContext is ALWAYS null.**

Surely there is something I can do?

**Comment by svrooij** [comment, marked as answer]
I thing you need to create a new class that implements the IHttpContextAccessor and allows you to return the http context.
